test/pytest/mocktransport.py

```diff
@@ -1,11 +1,55 @@
 import asyncio
 import json
+import os
+import subprocess
 
 from typing import Any, Dict, Iterable, Optional, Tuple
 
 MOCK_HOSTNAME = 'mockbox'
 
 
+def my_subprocesses():
+    try:
+        return subprocess.check_output(['pgrep', '-a', '-P', f'{os.getpid()}'], text=True).splitlines()
+    except subprocess.CalledProcessError:
+        # no processes found?  good!
+        return []
+
+
+def assert_no_subprocesses():
+    running = my_subprocesses()
+    if not running:
+        return
+
+    print('Some subprocesses still running', running)
+    # Clear it out for the sake of the other tests
+    subprocess.call(['pkill', '-9', '-P', f'{os.getpid()}'])
+    for _ in range(100):    # zombie vacuum
+        os.waitpid(-1, os.WNOHANG)  # will eventually throw, failing the test (which we want)
+    assert False  # more than 100?
```

Comment:
Just to be clear, as soon as `running` is non-empty, this should *always* fail, right?